design/setbehavior.api.md

```diff
@@ -0,0 +1,68 @@
+# Set Behavior API
+
+Currently behaviors are created as a part of the `setBehavior` call on Actor, and return back an immutable
+strongly typed behavior class of the behavior that is explicitly add through the call.  There is now a need
+to configure properties on the behavior before it is added to the actor and sent down to the client.  The 
+proposed changed is to deprecate the existing API for `setBehavior` and add in a new `setBehavior` that takes 
+an instance of a behavior.  This way the creation of the behavior can happen before it is set on the actor, 
+allowing for additional configuration by the developer.
+
+## API
+
+``` ts
+/**
+ * Sets the behavior on this actor.
+ * @param behavior The type of behavior to set. Pass null to clear any behaviors on the actor.
+ */
+public setBehavior(behavior: Behavior) {
+    if (behavior) {
+        this.internal.behavior = behavior;
+        this.context.internal.setBehavior(this.id, this.internal.behavior.behaviorType);
+        return;
+    }
+
+    this.internal.behavior = null;
+    this.context.internal.setBehavior(this.id, null);
+}
+```
+
+## Example Usage 
+
+***from input-test.ts***
+``` ts
+// Create an actor
+this.model = MRE.Actor.CreateFromGltf(this.assets, {
+    // from the glTF at the given URL, with box colliders on each mesh
+    uri: `${this.baseUrl}/monkey.glb`,
+    colliderType: 'box',
+    // Also apply the following generic actor properties.
+    actor: {
+        name: 'clickable',
+        parentId: root.id,
+        transform: {
+            local: {
+                scale: { x: 0.4, y: 0.4, z: 0.4 },
+                position: { x: 0, y: 1, z: -1 }
+            }
+        }
+    }
+});
+
+// Set up cursor interaction. We add the input behavior ButtonBehavior to the cube.
+// Button behaviors have two pairs of events: hover start/stop, and click start/stop.
+const buttonBehavior = new ButtonBehavior();
+buttonBehavior.onHover('enter', () => {
+    this.state = 1;
+    this.cycleState();
+});
+buttonBehavior.onButton('pressed', () => {
+    this.state = 2;
+    this.cycleState();
+});
+buttonBehavior.onHover('exit', () => {
+    this.state = 0;
+    this.cycleState();
+});
+
+this.model.setBehavior(buttonBehavior);
```

Comment:
What happens if I add the behavior to more than one actor?

Reply:
It would get call backs from each actor when the behavior was triggered.  This was one of the reasons why we initially did this more restrictive API, but it may actually be useful to have this behavior where you could have the `behavior` associated with more then one actor.  I am not sure that this restriction is necessary anymore, hence why I posted this API revision.  I would hope that a developer would understand how instances work in TS such that it would be reasonable expectation that if I gave and instance of something to multiple instances of an other thing, that both things would be acting on that instance.  Right?

Reply:
The question is: Is this a good API design? Behaviors are stateful. Is allowing multiple actors affect that state beneficial, innocuous, or harmful? I'm leaning toward harmful.

Reply:
We already have at least one other case that Steven is running in to where we want to be able to at the very least pass constructor arguments to the behavior.  I see your point on the potential harmful side effects that this can have, and agree that maybe this isn't the right approach.  Let me think on a better approach to solve this problem and I will put it in to another design PR after the tool design has finalized and I get that work in.